GAE/test/org/akvo/flow/api/app/DataPointServletTest.java

```diff
@@ -0,0 +1,205 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class DataPointServletTest {
+
+    private final LocalServiceTestHelper helper = new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+    @BeforeEach
+    public void setUp() {
+        helper.setUp();
+    }
+
+    @AfterEach
+    public void tearDown() {
+        helper.tearDown();
+    }
+
+
+    private DataPointAssignment createDataPointAssignment(Long assignmentId, Long deviceId, List<Long> dataPointIds, Long surveyId) {
+        final DataPointAssignment dpa = new DataPointAssignment();
+        final DataPointAssignmentDao dao = new DataPointAssignmentDao();
+
+        dpa.setDeviceId(deviceId);
+        dpa.setDataPointIds(dataPointIds);
+        dpa.setSurveyAssignmentId(assignmentId);
+        dpa.setSurveyId(surveyId);
+
+        return dao.save(dpa);
+    }
+
+    private SurveyAssignment createAssingment(Long surveyId, List<Long> deviceIds, List<Long> formIds) {
+        final SurveyAssignment sa = new SurveyAssignment();
+        final SurveyAssignmentDao dao = new SurveyAssignmentDao();
+
+        sa.setSurveyId(surveyId);
+        sa.setDeviceIds(deviceIds);
+        sa.setFormIds(formIds);
+
+        return dao.save(sa);
+    }
+
+    private List<SurveyedLocale> createDataPoints(Long surveyId, int howMany) {
+        final SurveyedLocaleDao dpDao = new SurveyedLocaleDao();
+        final List<SurveyedLocale> datapoints = new ArrayList<>();
+        for (int i = 0; i < howMany; i++) {
+            SurveyedLocale dataPoint = new SurveyedLocale();
+            dataPoint.setIdentifier(String.valueOf(i));
+            dataPoint.setSurveyGroupId(surveyId);
+            dataPoint.setDisplayName("dataPoint: " + i);
+            datapoints.add(dataPoint);
+        }
+        return new ArrayList<>(dpDao.save(datapoints));
+    }
+
+    private Set<Long> getEntityIds(List<? extends BaseDomain> entities) {
+        Set<Long> entityIds = new HashSet<>();
+        for (BaseDomain o : entities) {
+            entityIds.add(o.getKey().getId());
+        }
+        return entityIds;
+    }
+
+    @Test
+    public void datastoreReadWriteTest() {
+        final Long assignmentId = 1L;
+        final Long deviceId = 1L;
+        final List<Long> dataPointIds = Collections.singletonList(0L);
+        final Long surveyId = 1L;
+
+        createDataPointAssignment(assignmentId, deviceId, dataPointIds, surveyId);
+
+        final DataPointAssignmentDao dataPointAssignmentDao = new DataPointAssignmentDao();
+        final List<DataPointAssignment> dataPointAssignments = dataPointAssignmentDao.listByDeviceAndSurvey(deviceId, surveyId);
+        assertFalse(dataPointAssignments.isEmpty());
+
+        final DataPointAssignment assignment = dataPointAssignments.get(0);
+        assertEquals(assignment.getDataPointIds(), dataPointIds);
```

Comment:
The `assertEquals` params should be reversed I believe. `assertEquals(expected, actual)`